freqtrade/tests/finance/test_wallets.py

```diff
@@ -0,0 +1,32 @@
+# pragma pylint: disable=missing-docstring
+from freqtrade.tests.conftest import get_patched_freqtradebot
+from unittest.mock import MagicMock
+
+
+def test_sync_wallet_at_boot(mocker, default_conf):
+    mocker.patch.multiple(
+        'freqtrade.exchange.Exchange',
+        get_balances=MagicMock(return_value={
+            "BNT": {
+                "free": 1.0,
+                "used": 2.0,
+                "total": 3.0
+            },
+            "GAS": {
+                "free": 0.260739,
+                "used": 0.0,
+                "total": 0.260739
+            },
+        })
+    )
+
+    freqtrade = get_patched_freqtradebot(mocker, default_conf)
```

Comment:
i wouldn't rely on the `__init__` for the test, but rather test wallets.update() actively.
this does not test an update function - which needs to change the values ... ;)

remock with different return values below the asserts - and assert it's changed.... ;)